examples/ChatQnA/deploy/k8s_helm.md

```diff
@@ -0,0 +1,572 @@
+# Multi-node on-prem deployment with  TGI on Xeon Scalable processors on a K8s cluster using Helm Charts
+
+This deployment section covers multi-node on-prem deployment of the ChatQnA
+example with OPEA comps to deploy using the TGI service. There are several
+slice-n-dice ways to enable RAG with vectordb and LLM models, but here we will
+be covering one option of doing it for convenience : we will be showcasing  how
+to build an e2e chatQnA with Redis VectorDB and neural-chat-7b-v3-3 model,
+deployed on a Kubernetes cluster. For more information on how to setup a Xeon based Kubernetes cluster along with the development pre-requisites,
+please follow the instructions here (*** ### Kubernetes Cluster and Development Environment***). 
+For a quick introduction on Helm Charts, visit the helm section in  (**getting started**)
+
+## Overview
+
+There are several ways to setup a ChatQnA use case. Here in this tutorial, we
+will walk through how to enable the below list of microservices from OPEA
+GenAIComps to deploy a multi-node TGI megaservice solution.
+> **Note:** ChatQnA can also be deployed on a single node using Kubernetes, provided that all pods are configured to run on the same node.
+
+1. Data Prep
+2. Embedding
+3. Retriever
+4. Reranking
+5. LLM with TGI
+
+## Prerequisites
+
+### Install Helm
+First, ensure that Helm (version >= 3.15) is installed on your system. Helm is an essential tool for managing Kubernetes applications. It simplifies the deployment and management of Kubernetes applications using Helm charts. 
+For detailed installation instructions, please refer to the [Helm Installation Guide](https://helm.sh/docs/intro/install/)
+
+### Clone Repository 
+First step is to clone the GenAIInfra which is the containerization and cloud native suite for OPEA, including artifacts to deploy ChatQnA in a cloud native way.
+
+```bash
+git clone https://github.com/opea-project/GenAIInfra.git
+```
+Checkout the release tag
+```
+cd GenAIInfra/helm-charts/
+git checkout tags/v1.0
+```
+### HF Token
+The example can utilize model weights from HuggingFace and langchain.
+
+Setup your [HuggingFace](https://huggingface.co/) account and generate
+[user access token](https://huggingface.co/docs/transformers.js/en/guides/private#step-1-generating-a-user-access-token).
+
+Setup the HuggingFace token
+```
+export HF_TOKEN="Your_Huggingface_API_Token"
+```
+
+### Proxy Settings
+Make sure to setup Proxies if you are behind a firewall.
+For services requiring internet access, such as the LLM microservice, embedding service, reranking service, and other backend services, proxy settings can be essential. These settings ensure services can download necessary content from the internet, especially when behind a corporate firewall.
+Proxy can be set in the `values.yaml` file, like so:
+Open the `values.yaml` file using an editor
+```bash
+vi GenAIInfra/helm-charts/chatqna/values.yaml
```

Comment:
In the previous step we did a cd into the GenAIInfra/helm-charts directory so this command would fail.